### ISSUE TITLE
False positive upgrading to OCaml 5.4 for annotated-ignore

### ISSUE THREAD

**Issue by mbarbin** [janestreet/ppx_js_style#4]
Upgrading to OCaml 5.4 with ppx_js_style version v0.17.1 I get the following error:

```ocaml
File "doc/docs/guides/usage-styles/lib/usage_styles.ml", lines 3-8, characters 2-8:
3 | ..Command.make
4 |     ~summary:"A command skeleton."
5 |     (let open Command.Std in
6 |      let+ (_ : int) = Arg.named [ "n" ] Param.int ~doc:"A value for n."
7 |      and+ () = Arg.return () in
8 |      ())
Error: Jane Street style: Ignored expression must come with a type annotation
```

This is a new error, which doesn't fire with OCaml 5.3. There is not unannotated ignore expression here, so the expectations is that the linter should *not* raise this altert.

I am not exactly sure where this comes from. This is the first issue I open about this error, starting from the ppx_js_style project itself, but this may very well be due to an interaction between several projects and dependencies.

This is from the repo [cmdlang](https://github.com/mbarbin/cmdlang/tree/main/doc/docs/guides/usage-styles/lib) - I think just trying to build all targets in that repo with a 5.4 switch should be enough to try and reproduce the error.

Thanks!

**Comment by nmatschke** [comment]
I was unable to reproduce this issue on a 5.4 switch.

**Comment by nmatschke** [comment]
I would make sure you're using the latest `ppxlib` and `ppxlib_jane` possible.
